contracts/cw3-dao/src/state.rs

```diff
@@ -128,27 +121,112 @@ impl Proposal {
                 if self.expires.is_expired(block) {
                     // If expired, we compare vote_count against the total number of votes (minus abstain).
                     let opinions = self.votes.total() - self.votes.abstain;
-                    vote_count >= votes_needed(opinions, threshold)
+                    self.votes.yes >= votes_needed(opinions, threshold)
                 } else {
                     // If not expired, we must assume all non-votes will be cast against
-                    // vote_count
                     let possible_opinions = self.total_weight - self.votes.abstain;
-                    vote_count >= votes_needed(possible_opinions, threshold)
+                    self.votes.yes >= votes_needed(possible_opinions, threshold)
                 }
             }
         }
     }
 
-    /// returns true iff this proposal is sure to pass (even before expiration if no future
-    /// sequence of possible votes can cause it to fail)
-    pub fn is_passed(&self, block: &BlockInfo) -> bool {
-        self.does_vote_count_reach_threshold(self.votes.yes, block)
-    }
-
-    /// As above for the rejected check, used to check if a proposal is
-    /// already rejected.
+    /// Returns true if this proposal is sure to be rejected (even before expiration, if
+    /// no future sequence of possible votes could cause it to pass).
     pub fn is_rejected(&self, block: &BlockInfo) -> bool {
-        self.does_vote_count_reach_threshold(self.votes.no, block)
+        match self.threshold {
+            Threshold::AbsolutePercentage {
+                percentage: percentage_needed,
+            } => {
+                let options = self.total_weight - self.votes.abstain;
```

Comment:
As discussed in the Discord, let's add a special case for if `options` is zero. This will reject proposals in the all abstain case.

Reply:
Otherwise this is looking really great. :) Appreciate your help iterating on this threshold stuff.